src/HIE/Bios/Config.hs

```diff
@@ -127,16 +130,28 @@ parseCabal = parseStackOrCabal Cabal CabalMulti
 parseBios :: Value -> Parser (CradleType a)
 parseBios (Object x)
     | 2 == Map.size x
-    , Just (String biosProgram) <- Map.lookup "program" x
-    , Just (String biosDepsProgram) <- Map.lookup "dependency-program" x
-    = return $ Bios (T.unpack biosProgram) (Just (T.unpack biosDepsProgram))
+    , Just biosCallable <- exclusive (stringTypeFromMap Program "program") (stringTypeFromMap Command "shell")
+    , Just biosDepsCallable <- exclusive (stringTypeFromMap Program "dependency-program") (stringTypeFromMap Command "dependency-shell")
```

Comment:
Looks like here you could have `program` and `dependency-shell` which I suppose is acceptable?

Reply:
Yes, that was intentional. I didn't see a good reason to disallow it, and I added a test and documentation explicitly for this kind of mixing.